src/main/java/org/broadinstitute/hellbender/tools/walkers/gnarlyGenotyper/GnarlyGenotyperEngine.java

```diff
@@ -345,19 +350,30 @@ else if (g.countAllele(Allele.NON_REF_ALLELE) > 0) {
 
     /**
      * For a genotype with likelihoods that has a no-call GT, determine the most likely genotype from PLs and set it
+     * We use a GenotypeLikelihoodCalculator to convert from the best PL index to the indexes of the alleles for that
+     * genotype so we can set the GenotypeBuilder with the alleles
      * @param gb    GenotypeBuilder to modify and pass back
-     * @param genotypeLikelihoods   PLs to use to call genotype
-     * @param allelesToUse  alleles in the parent VariantContext, because GenotypeBuilder needs the allele String rather than index
+     * @param genotypeLikelihoods   PLs to use to call genotype; count should agree with number of alleles in allelesToUse
+     * @param allelesToUse  alleles in the parent VariantContext (with ref), because GenotypeBuilder needs the allele String rather than index
      */
-    private void makeGenotypeCall(final GenotypeBuilder gb,
+    @VisibleForTesting
+    protected void makeGenotypeCall(final GenotypeBuilder gb,
                                         final double[] genotypeLikelihoods,
                                         final List<Allele> allelesToUse) {
 
         if ( genotypeLikelihoods == null || !GATKVariantContextUtils.isInformative(genotypeLikelihoods) ) {
             gb.alleles(GATKVariantContextUtils.noCallAlleles(ASSUMED_PLOIDY)).noGQ();
         } else {
             final int maxLikelihoodIndex = MathUtils.maxElementIndex(genotypeLikelihoods);
-            final GenotypeLikelihoodCalculator glCalc = glcCache.get(allelesToUse.size());
+
+            GenotypeLikelihoodCalculator glCalc;
+            if ( allelesToUse.size() <= maxAltAllelesToOutput ) {
```

Comment:
`allelesToUse` includes the ref allele, yet we are testing for equality against `maxAltAllelesToOutput`. Can you add a comment explaining why this is correct?

Reply:
It's not, so I did what you suggested and made a `maxAllelesToOutput` variable.  In an ideal world, maybe it would make sense to have a wrapper class for a list of Alleles that had containsRef() and containsNonRef() methods.